config/main.py

```diff
@@ -6594,6 +6630,40 @@ def sample_rate(ctx, ifname, rate):
             except ValueError as e:
                 ctx.fail("Invalid ConfigDB. Error: {}".format(e))
 
+#
+# 'sflow' command ('config sflow interface sample-direction  ...')
+#
+@interface.command('sample-direction')
+@click.argument('ifname', metavar='<interface_name>', required=True, type=str)
+@click.argument('direction', metavar='<sample_direction>', required=True, type=str)
+@click.pass_context
+def sample_direction(ctx, ifname, direction):
+    config_db = ValidatedConfigDBConnector(ctx.obj['db'])
+    if ADHOC_VALIDATION:
+        if not interface_name_is_valid(config_db, ifname) and ifname != 'all':
+            click.echo('Invalid interface name')
+            return
+        if direction:
+            if direction not in ['rx', 'tx', 'both']:
+                ctx.fail("Error: Direction {} is invalid".format(direction))
+
+            if (direction == 'tx' or direction == 'both') and (is_port_egress_sflow_supported() == 'false'):
+                ctx.fail("Sample direction {} is not supported on this platform".format(direction))
+
+    sess_dict = config_db.get_table('SFLOW_SESSION')
```

Comment:
What if 'SFLOW_SESSION' does not exist?

Reply:
If SFLOW_SESSION does not exits, it will add the entry at line 6662 "config_db.mod_entry('SFLOW_SESSION', ifname, {'sample_direction': direction})"